main.go

```diff
@@ -171,14 +163,16 @@ func main() {
 // Get the client's real IP address using the X-Real-IP header, or if that is
 // empty, http.Request.RemoteAddr. See the sample nginx.conf for using the
 // real_ip module to correctly set the X-Real-IP header.
-func getClientIP(r *http.Request) string {
+func getClientIP(r *http.Request) (string, error) {
 	xRealIP := r.Header.Get("X-Real-IP")
```

Comment:
i don't know why we are even looking at this header. what's wrong with just r.RemoteAddr?